### PR TITLE
feat(local-debug): terminate all running tasks when error occurs

### PR DESCRIPTION
**Pull request by kuojianlu**
when error occurs during pre debug check, terminate all running teamsfx tasks and send error telemetry.

### CODE REVIEW

**Review by jayzhang — APPROVED** [review]

File: packages/vscode-extension/src/debug/teamsfxTaskHandler.ts [hunk 1]
```diff
@@ -79,6 +79,16 @@ function onDidStartDebugSessionHandler(event: vscode.DebugSession): void {
     }
 }
 
+export function terminateAllRunningTeamsfxTasks(): void {
+  for (const task of allRunningTeamsfxTasks) {
```
Comment (swatDong) [inline packages/vscode-extension/src/debug/teamsfxTaskHandler.ts:83]: clear `allRunningTeamsfxTasks`?

Comment (kuojianlu) [reply packages/vscode-extension/src/debug/teamsfxTaskHandler.ts:83]: Good catch. Updated.

File: packages/vscode-extension/src/debug/teamsfxTaskHandler.ts [hunk 2]
```diff
@@ -93,17 +104,10 @@ function onDidTerminateDebugSessionHandler(event: vscode.DebugSession): void {
...
-                    process.kill(task[1], "SIGINT");
-                } catch(e) {
-                    // ignore and keep killing others
-                }
-            }
+            terminateAllRunningTeamsfxTasks();
         }
 
         allRunningDebugSessions.delete(event.id);
-        allRunningTeamsfxTasks.clear();
```
Comment (swatDong) [inline packages/vscode-extension/src/debug/teamsfxTaskHandler.ts:106]: Do not remove this...

Comment (kuojianlu) [reply packages/vscode-extension/src/debug/teamsfxTaskHandler.ts]: Sure. Added it back.

**Review by swatDong — APPROVED** [review]

File: packages/vscode-extension/src/resources/Strings.json [hunk 3]
```diff
@@ -105,6 +105,10 @@
     "graphLogin": {
       "warningMsg": "The Teams Toolkit requires a Microsoft 365 account. This is the account that you use to log in to Microsoft Teams. The Teams Toolkit will use this account to manage applications and service principals."
     },
+    "localDebug": {
+        "portAlreadyInUse": "Port: %s is already in use. Close this port and try again.",
+        "portsAlreadyInUse": "Ports: %s are already in use. Close these ports and try again."
+    },
```
Comment (1openwindow) [inline packages/vscode-extension/src/resources/Strings.json:111]: @timngmsft for awareness

**Review by 1openwindow — APPROVED** [review]

**Pull request merged**